### ISSUE TITLE
int8 give wrong results

### ISSUE THREAD

**Issue by Dapid** [serge-sans-paille/pythran#822]
Consider this program_
```
import numpy as np

#pythran export compute(int8[:])
def compute(x):
    print(np.max(x))
    print(np.min(x))
    print(x.max())

    return x.max()
```

Call with:
```
import numpy as np
import a
o = a.compute(np.random.randint(0, 3, size=20).astype(np.int8))
```

On the released version of Pythran, the terminal output is mysterious:
```



```
but it rightfully returns a 2.

On Pythran master, it returns -128. The screen output is:

```
-128
127
-128
```

**Comment by Dapid** [comment]
It seems that the problem happens only with min/max, but x[0] is correct.

**Comment by serge-sans-paille** [comment]
Got it, and the fix is under testing :-)

**Comment by serge-sans-paille** [comment]
#823 should do the trick!

**Comment by serge-sans-paille** [comment]
Closed by #823
